app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -1002,12 +1003,14 @@ export default Controller.extend(bufferedProperty("model"), {
     },
 
     changeNotice(post) {
+      const modal = this.modal;
       return new Promise(function (resolve, reject) {
-        const modal = showModal("change-post-notice", { model: post });
-        modal.setProperties({
-          resolve,
-          reject,
-          notice: post.notice ? post.notice.raw : "",
```

Comment:
I moved this into the component, since we're already passing the post, and can grab the notice from it internally.